src/constants/tokens.ts

```diff
@@ -0,0 +1,40 @@
+import { Token, NativeCurrency, Currency, WETH9 } from '@uniswap/sdk-core'
+import { SupportedChainId } from '../constants/chains'
+import { CowError } from '../utils/common'
+import { WXDAI, XDAI_NAME, XDAI_SYMBOL } from './xdai'
+
+export const WETH9_EXTENDED: { [chainId: number]: Token } = {
+  ...WETH9,
+  [SupportedChainId.GNOSIS_CHAIN]: WXDAI,
+}
+
+export class GpEther extends NativeCurrency {
```

Comment:
Agree with Anxo, we better duplicate a mapping of networks and native wrapped currency addresses to avoid dependency on uniswap sdk.